### PR TITLE
Add CodeSandbox interactive demo (with Next.js)

### PR DESCRIPTION
**Pull request by karlhorky**
Hi @octref ! 👋

I have been considering different highlighting options (also Prism.js with `react-syntax-highlighter`), and stumbled upon your project through Orta (who used it for the [`shiki-twoslash`](https://www.npmjs.com/package/shiki-twoslash) package for the TypeScript website)

One thing that I was looking for when going through this repo and also the docs website was an interactive demo.

I've set one up on CodeSandbox (with Next.js). It is currently a frozen sandbox (so that I remember to not accidentally make changes to it), but of course happy to have you fork it so that you control the sandbox.

What are your thoughts?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <h2 align="center">Shiki</h2>
 </p>
 <p align="center">
-  Shiki is a beautiful Syntax Highlighter. <a href="http://shiki.matsu.io">Demo</a>.
+  Shiki is a beautiful Syntax Highlighter. <a href="http://shiki.matsu.io">Docs</a>. <a href="https://codesandbox.io/s/shiki-next-js-cir0y">CodeSandbox</a>.
```
Comment (octref) [inline README.md:5]: I wouldn't want to change this, sorry.

**Comment by octref** [comment]
> of course happy to have you fork it so that you control the sandbox

Sure, I'll fork and make a sandbox. Thanks for bringing this up. Making it easier for people to fiddle with the API would be great for trying it out and bug reporting.

**Review by octref — CHANGES_REQUESTED** [review]

**Comment by octref** [comment]
@karlhorky It's all good but I'll fork and make a few more necessary changes before merigng. Nothing you need to do on your side. Thanks 👍

File: README.md (hunk 1 above)
Comment (karlhorky) [reply README.md:5]: All good, just wanted to suggest surfacing it somewhere prominent to save users time.

Other solutions would be good too!

**Comment by octref** [comment]
I didn't have much success with codesandbox, not sure what's wrong: https://codesandbox.io/s/pmkh0?file=/style.css

image


I can only accept setup like this. I don't want to bring in a lot of dependencies: https://github.com/octref/shiki-starter
If CodeSandbox can do this, I can link to it. Otherwise I'd rather leave a link to the starter template.

**Comment by karlhorky** [comment]
@octref Yeah I also ran into this - that's why I switched to Next.js, where the dependency resolution seems to work properly.

However, if you want fewer dependencies, looks like it also works just with plain Node.js:

https://codesandbox.io/s/relaxed-haslett-cpuxm?file=/src/index.js

**Comment by octref** [comment]
CodeSandbox seems really flakey...I can't even move `src/index.js` to `index.js` otherwise it would error out. I cannot change `start` script or `nodemon` devDependency or it would fail:

image

I'm not 100% happy about CSB, but I do agree having a click-to-play option would be nice until something better is available:
https://codesandbox.io/s/shiki-starter-gphup

**Review by octref — APPROVED** [review]

**Comment by octref** [comment]
Thanks for doing this 🌟

**Comment by karlhorky** [comment]
> CodeSandbox seems really flakey...I can't even move `src/index.js` to `index.js` otherwise it would crash. I cannot change `start` script or `nodemon` devDependency or it would fail

> I'm not 100% happy about CSB

Ah, you're right, CodeSandbox is not normally this bad, not sure what's up. I've opened two issues about the problems I've seen:

1. https://github.com/codesandbox/codesandbox-client/issues/4806
2. https://github.com/codesandbox/codesandbox-client/issues/4807

---

I've tried repl.it, and it seems to be much more stable:

https://repl.it/@karlhorky/PoliticalPricklyBase#index.js

**Comment by karlhorky** [comment]
@octref I've tried repl.it, and it seems to be much more stable.

Maybe this is a better option for the demo:

repl.it/@karlhorky/PoliticalPricklyBase#index.js

**Comment by octref** [comment]
@karlhorky Thanks, I have created https://repl.it/@octref/shiki-starter and linked to it.

**Comment by karlhorky** [comment]
Awesome, looks great, thanks!

**Comment by lbogdan** [comment]
Hello, there! Sorry for the poor UX you experienced with CodeSandbox Containers. The way it works is we look for a `start` (or `dev`, `devel`, `serve`) script in `package.json` which we start inside the container, we expect that's a server that listens on one (or more) port(s), and then we expose those ports on `https://id(-port).sse.codesandbox.io`. 

Now, whenever you change the scripts in `packge.json`, you need to restart the sandbox with the "Restart Sandbox" button in the "Server Control Panel" side tab, and we stop the running server and start the new one.

image

I've made a simple starter with `index.js` under the project root here: https://codesandbox.io/s/shiki-starter-gqcfl?file=/index.js .

**Comment by octref** [comment]
@lbogdan Thanks! I tried refreshing to restart the sandbox. Didn't thought I need to press another button.

**Pull request merged**